sdcm/logcollector.py

```diff
@@ -0,0 +1,849 @@
+        CommandLogEntity(name='aalert.log',
+                         command='docker logs --details -t aalert'),
+        FileLogEntity(name='scylla_manager.log',
+                      command='sudo journalctl -u scylla-manager.service --no-tail',
+                      search_locally=True),
+        PrometheusSnapshotsEntity(name='prometheus_data'),
+        MonitoringStackEntity(name='monitoring-stack'),
+        GrafanaScreenShotEntity(name='grafana-screenshot'),
+        GrafanaSnapshotEntity(name='grafana-snapshot')
+    ]
+    cluster_log_type = "monitor-set"
+
+
+class SCTLogCollector(LogCollector):
+    """logs for hydra test run
+
+    Find and collect local version of files
+    for sct
+
+    Extends:
+        LogCollector
+
+    """
+    log_entities = [
+        FileLogEntity(name='sct.log',
+                      search_locally=True),
+        FileLogEntity(name='raw_events.log',
+                      search_locally=True),
+        FileLogEntity(name='events.log',
+                      search_locally=True),
+        FileLogEntity(name='output.log',
+                      search_locally=True),
+        FileLogEntity(name='critical.log',
+                      search_locally=True)
+    ]
+    cluster_log_type = 'sct-runner'
+
+    def collect_logs(self, local_search_path=None):
+        for ent in self.log_entities:
+            ent.collect(None, self.local_dir, None, local_search_path=local_search_path)
+        if not os.listdir(self.local_dir):
+            LOGGER.warning('No any local files')
+            return self.local_dir, None, None
+
+        final_archive = self.archive_dir_with_zip64(self.local_dir)
+
+        s3_link = self.upload_logs(final_archive, "{0.test_id}/{0.current_run}".format(self))
+        remove_files(self.local_dir)
+        remove_files(final_archive)
+        return s3_link
+
+    def archive_dir_with_zip(self, logdir):
+        return shutil.make_archive("{}".format(logdir), "zip", root_dir=logdir)
+
+    def archive_dir_with_zip64(self, logdir):
+
+        archive_base_name = os.path.basename(logdir)
+        archive_storing_dir = os.path.dirname(logdir)
+        archive_full_name = os.path.join(archive_storing_dir, archive_base_name + ".zip")
+        cur_dir = os.getcwd()
+        try:
+            with zipfile.ZipFile(archive_full_name, "w", allowZip64=True) as arch:
+                os.chdir(logdir)
+                for root, _, files in os.walk(logdir):
+                    for log_file in files:
+                        arch.write(log_file)
+                os.chdir(cur_dir)
+        except Exception as details:
+            LOGGER.error("{}".format(details))
+            archive_full_name = None
+        return archive_full_name
+
+
+class Collector:
+    """Collector instance
+
+    Collector instance which should be run to collect logs and additional info
+    as separate stage in pipeline on as subcommand in hydra
+
+    """
+    LOCAL_SEARCH_FOR_USER = "jenkins"
```

Comment:
I still don't understand what would happen if I run this on my computer after I've run a test:
```
hydra collect-logs `cat ~/sct-results/latest/test_id`
```

And what would happen if I've run from my computer 
```
# where test_id is a from a job running in jenkins
hydra collect-logs [test_id]
```

Reply:
implemented. new command return builder and folder of testrun by test-id.
limitation is that builders are hardcoded